### ISSUE TITLE
SetBreakpointsRequest accounts for tab size in column

### ISSUE THREAD

**Issue by hinshun** [microsoft/vscode#110988]
- VSCode Version: 1.51.1
- OS Version: Linux x64 5.4.0-48-generic

Steps to Reproduce:

1. While debugging via DAP extension, highlight a range to set an inline breakpoint
2. Tabs considered as single column in most APIs, but for `SetBreakpointsRequest`, its considered `<tabsize>`.

Does this issue occur when all extensions are disabled?: Yes/No
Yes

---

For most of DAP's API, I have been sending back source locations assuming tabs counts as a single column and VS Code seems to accept them correctly (and taking into account that `\t  == 1 column`). However, I noticed that VS Code shows the column as if tabs were `<tabsize>` columns, and also sends that column for inline `SetBreakpointsRequest`.

![tabsize](https://user-images.githubusercontent.com/6493975/99744798-3b628600-2a8d-11eb-9d86-c9740e45fab2.png)

^ For other editors and from my parser's point of view, this is usually `Ln 7, Col 2`.

I couldn't find an option to change this assumption, is this the intended behavior?

**Comment by weinand** [comment]
Since debug adapters don't know the tab size, DAP's column positions are really character positions.
So there is no difference in the column position of the word "foo" in the two strings `"\tfoo"` and `" foo"`.  

@isidorn can you please check that VS Code is doing this?

**Comment by isidorn** [comment]
I just checked and VS Code counts a tab as one character.
This can easily be seen in my gif where I am pressing tab and the column number in the Breakpoints view increases by 1, not by 4 even though I have tab size set to 4. And that exact number from Breakpoints view will also be sent to the Debug Adapter.

@hinshun Please note:
1. Columns start with 1, not 0
2. Your editor is probably configured to indent using spaces, so you press tab but actually 4 spaces are indented. `F1 > Convert indentation to Tabs`, to be sure you are using tabs

![recording (1)](https://user-images.githubusercontent.com/1926584/99831908-14c14f80-2b60-11eb-91a3-c5faa5729ef3.gif)

**Comment by hinshun** [comment]
Hi @isidorn, sorry for the noise, I can confirm that it's sending correctly.

I'm new to setting inline breakpoints so I started by: Double clicking a `[token]` -> Right click -> Set inline breakpoint, which causes the breakpoint column to be set at the column after the `[token]` so technically the breakpoint is not within that node causing it defaulted to a line breakpoint.

TLDR: Intuitively I was trying to set a breakpoint for that `[token]` but it ended up setting it after it.

**Comment by isidorn** [comment]
@hinshun Thanks for the explanation and for letting us know.
